### DISCUSSION TITLE
Steering Committee - Meeting Minutes and Decisions

### DISCUSSION THREAD

**Discussion by binste** [vega/.github#7] [Meeting Notes]
We'll use this discussion to document decisions taken in meetings of the [Vega Steering Committee](https://github.com/vega/.github/blob/main/STEERING-COMMITTEE.md). Everyone is welcome to provide inputs by replying to the individual discussion comments. Top-level comments are ideally to be used for the minutes only.

See https://github.com/altair-viz/altair/discussions/3164 for a history of the meeting minutes and decisions of the former Vega-Altair Steering Committee which got merged into the Vega Steering Committee.

**Comment by binste** [comment]
# 2024-03-15

Participants:

- @binste
- @joelostblom 
- @domoritz 
- @jonmmease 
- @mattijn

# Decisions and next actions
* The consensus was that we want to move all active repositories from https://github.com/altair-viz to https://github.com/vega. The main challenge we identified will be to ensure redirects from the current Altair docs page to the new one after the move.
  * As preparation, @binste will set up a test repo for the docs in altair-viz to redirect to another test repo in Vega. The redirects need to be on every single page of the docs.
  * The repo `vega_datasets` exists in both organizations. We'll rename the one from altair-viz to `python_vega_datasets`. Later on, these two could be merged into one.
* Altair docs now use plausible.io, sponsored by scientific-python.org, for privacy-friendly web analytics. See https://github.com/altair-viz/altair/pull/3346. This could also be interesting for the Vega and Vega-Lite docs.
  * @binste to contact folks at scientific-python.org if they'd be ok with adding the Vega and Vega-Lite docs to their plausible.io server for privacy-friendly web analytics.  for the corresponding PR in the Altair docs
  * @binste to invite @domoritz to Altair Plausible space so Dominik can see what kind of data is available
* We've decided by consensus that we want to apply as a "Vega organization" for NumFOCUS Fiscal Sponsorship, deadline 15.04.
  * @joelostblom to set up a Google Docs and start drafting the application. Logos available at https://github.com/vega/logos if needed. We might also need to update the https://github.com/vega/roadmap for the application or use the Altair roadmap for now.
* All participants (majority of steering committee) voted in favour of accepting [VegaFusion](https://github.com/hex-inc/vegafusion) to the Vega organization. Big thank you to [hex-inc](https://github.com/hex-inc) for donating the project and to @jonmmease for all the hard work that he put into it 🥳 
  * @jonmmease to move the VegaFusion repo to the Vega org and add the project to the beginning of the https://github.com/vega/.github/blob/main/CHARTER.md document
* Not a "next action" but an idea we discussed was to document the various ways how students could get involved in the Vega projects such as “If you’re part of university X, ... or consider applying for Google Summer of Code”.

**Comment by domoritz** [comment]
# 2025-02-17

Participants

* @domoritz 
* @arvind 
* @kanitw 
* @jonmmease 
* @mattijn 
* @joelostblom 
* @hydrosquall (note-taker)

## Agenda

* Welcome for Cameron
* NumFocus
  * Tabled discussions for next time: linking sponsorship to our internal roadmap, role corporate sponsors (Hex, DB, DD) might play
* Updates from Altair
  * @mattijn mentioned a Basemaps / signal linking challenge
  * @jonmmease / @joelostblom gauging feasibility of presentation (PyCon, PyData, Scipy) at this summer's events
  * Note presentations at conferences affect NumFocus community health scores
* Do we want two chairs (Vega and Altair)?
* Security updates (Cameron)
  * We have an [org-wide security advisory policy](https://docs.github.com/en/code-security/getting-started/adding-a-security-policy-to-your-repository). Specific repos can [add their own](https://github.com/vega/vega/blob/main/SECURITY.md). Encourage new users to use this instead of public github issues.
  * The vega security backlog has been cleared over the past quarter so we have a clean slate!
  * Future XSS reports should be tested with the [expressionInterpreter mode](https://github.com/vega/editor/pull/1461) in editor to gauge impact (severity is higher if the exploit works even with interpreter enabled)
* NumFocus code of conduct: https://numfocus.org/code-of-conduct
* ESM updates and next release planning (major releases for Vega-Lite)
  * Migrating our infra will make build maintenance easier / enable better testing/dependency management
  * Recommending dropping CJS, keeping UMD + ESM
  * Discussion around changing renderer defaults (vega-embed to prefer SVG for a11y, altair to prefer canvas for performance)
  * Discussion around feasibility of bringing vega-label / lasso selections to 6.x release w/ Ham and Arvind

## Decisions and next actions

* Numfocus / Sponsors
  * @domoritz to add Numfocus mentions to relevant README.md / docs sites
  * @domoritz to look into Vega Pro [team slack (NF portal)](https://numfocus.happyfox.com/kb/article/16-get-a-slack-pro-account/)
  * @mattijn to look into LinkedIn profile
  * Team to cross-promo Numfocus partnerships on Bluesky, LinkedIn, maybe X if applicable
* Two chairs?
  * Primary role of chair is scheduling meeting, make agenda - responsibility is shared among steering committee for ownership
  * We'll keep one for now, but Dom can flag if workload increases/splitting is helpful
  * @domoritz to set a reminder to send reminder day before event to account for Gcal spam filters
* Security
  * Discussed merits of sync expression interpreter state to URL params, declined for now
  * Improve credentials management (discussed in slack)
* Numfocus Code of Conduct
  * @hydrosquall to look into difference between [Numfocus](https://numfocus.org/code-of-conduct) and existing Vega [code of conduct](https://github.com/vega/.github/blob/main/CODE_OF_CONDUCT.md), will share w/ Committee if there are meaningful differences
 * ESM and next release
  * Targeting Mar 14 2025 release
  * @domoritz will be driving the ESM migration
  * Further milestones in https://github.com/vega/vega-lite/milestone/72

**Reply by mattijn** [reply]
Created a LinkedIn profile: https://www.linkedin.com/company/vega-viz and added steering-committee members as admins, please let me know if I missed someone.

**Comment by domoritz** [comment]
# 2025-05-27

Participants

@domoritz
@kanitw
@jonmmease
@mattijn (note-taker)
@joelostblom
@n8agrin
@hydrosquall
@cmerrick

## Agenda

* Vega conference (Omni is willing to sponsor and help with logistic). CC @cmerrick @n8agrin
* Proposals for project funding (see https://github.com/vega/.github/discussions/25)
* NumFocus checking meeting
* Funding + Sponshorships, open collective
* User study request

## Notes and Decisions

* Jon will attend SciPy

**Participating in Studies**

* What’s our view? What extra work does it bring?
* Only join after agreement in the steering committee.

**Funding Use**

* Use the simple proposal template: [https://github.com/vega/.github/discussions/25](https://github.com/vega/.github/discussions/25)
* Cameron: Review frequency? → For now, review proposals when they come in.

**Open Collective**

* Vega is on Open Collective: [https://opencollective.com/dashboard/vegaprojects/overview](https://opencollective.com/dashboard/vegaprojects/overview)
* Let the use of funds grow naturally. Collect some examples.
* Examples: support infra, help finish a PR, general community health.
* No promises. It’s for the project’s basic needs.
* The steering committee decides how it’s used.

**Funding Notes**

* Only use for non-profit goals.
* If features are delayed, be open about it.
* Use free Cloudflare account for now.

**Community Growth**

* Keep it informal. People now grow into roles by doing.
* Christopher: Write where people can join, and tag them on Slack.
* Goal: A growing group with different roles.

**Vega Conf**

* Last time: Seattle. North America still makes sense.
  Options: MIT (Boston), UBC (Vancouver), NYC, Seattle.
* Omni wants to help organize.
* Target: developers and experienced users.
* Christoph: Community health is a goal itself.
* Developers are key, but advanced users matter too.
* Dom: Share stories like “how we use Vega-Lite.”
* Christoph: University is a good place. Make a shortlist.
* Last conf: 20–30 people. This time: aim for 50+.
* Invite-only or via light application (maybe with donation).
* Should we ask for tickets or donation?
* Create a small team to work on this.
* Can we do it in the fall?
  Check overlap with:

  * [https://ieeevis.org/year/2025/welcome](https://ieeevis.org/year/2025/welcome)
  * [https://www.outlierconf.com/](https://www.outlierconf.com)
  * [https://odsc.com/california/](https://odsc.com/california/)
* Joining another conference? Could be a good idea.
* Goal of the event:

  * Bring together steering committee, devs, maintainers
  * Involve experienced users
  * Give clear direction (roadmap: [https://github.com/orgs/vega/projects/9/views/3](https://github.com/orgs/vega/projects/9/views/3))
  * Make it worth attending

**Lightning Talk Ideas**

* Altair Express
* Vega to ESM
* How companies use Vega
* Ideas for roadmap
* Distributed workflows

**To Do**

* NumFocus: Ask if we can use funding for work on features.
* Cameron & Jon: Write proposal for the Vega org README
 and add it to the org.
* Conf team: Dom, Chris, Nathan, Kanit

**Comment by domoritz** [comment]
# 2025-08-27

Participants:

- @ChristopherDavisUCI 
- @joelostblom
- @domoritz
- @jonmmease
- @kanitw

## Agenda

- Vega conference planning
- Encourage faster turnaround times on pull requests

## Notes and Decisions

### Vega conference 

- Date set! October 14-15, same location in Seattle as last time. Edit: cancelled
- There will be more focus on users of the Vega ecosystem this time and room to share user stories.
- We will encourage more users to contribute as developers and probably have a session on how to become a maintainer.
- Will try to advertise more this time since the focus is on users too: both post on social channels and as issues in the main repos. 
- Some sponsorship will be sought, but not to support full accommodation and travel.
- Tickets will be sold, mostly to get commitment from participants rather than try to support the conference financially.

### Faster turnaround on PR and issues

- We want to avoid bottlenecks on single people in the repos, especially when a single person is responsible for a lot of PRs which becomes overwhelming.
- People are already busy at work so it's hard to get more commitment for PR reviews etc.
- Maybe we should use a code owners system where specific maintainers are responsible for specific parts of the code and any PRs opened against those parts.
- Another option would be to have a lead maintainer who takes more organizational responsibilities and suggests other maintainers to review when PRs come in. 
- Maybe having a short weekly/bi-weekly meeting for repo-specific PRs could help talk through roadblocks and keep movement constant. Dom and Ham will start for Vega-Lite.
- There could also be some recurring meetings where new people could join and get help in contributing to the ecosystem. This is a bigger commitment and would require someone to volunteer to lead them. We probably don't have resources for this right now.

### Numfocus

- We have received some sponsorship already, but can maybe advertise more widely to get the word out that we are now accepting donations to the project.
- What we got so far has helped sponsor a rewrite of the vega editor: enabling logins, adding unit tests, allowing linking to vega lite and vega locally to help development, and more. 

### Misc

- A new version of Vega react was just merged, which makes it more lightweight and up to date. 
- Altair tutorial at scipy. 
- Vega lite 6 update is the next priority

**Comment by domoritz** [comment]
# 2025-11-18

## Participants

@joelostblom
@mattijn 
@domoritz 
@jonmmease 

## Agenda

### Updates from Altair, Vega, etc
* Altair v6 release. Went smoothly (slight delay for Python 3.14 release). No release specific bug reports. Time encoding channel not specifically called out (pending docs updates in Vega-Lite). 
* Vega v5 security release. We did a re-release of a previous version for the first time to address a security issue. Generally want to avoid it in the future.

### Funding

* Could set up funding documents with tiers and specific offers.
* Dom could use funding to pay students. Committee supports using funding for that.
* Should maybe outline what we want to do with the funding.
  * Fund students to help with specific feature
  * Raise money for maintainer
  * Potential features: MCP, Roadmap, etc???
* Other things we can offer right now (tiers)
  * Name or logo on https://vega.github.io
  * Slot on maintainer meeting (once we have it)
  * Later: get input on features
  * Later: set up a bounty system or engage with a developer or org like quansight for feature development
* @mattijn and @domoritz will draft a document and get feedback from steering committee

### How to push PRs over the line

* Questions about region selection pull request: https://github.com/vega/vega-lite/pull/9536
* Discussion of Animation feature. @domoritz is trying to push the last bit @jonathanzong and @arvind need to get the pull request ready.
* Idea: start a regular maintainer meeting (should be open). Sponsors get a block for telling people about what they do, why the feature is important. Could be run by a maintainer person.

**Reply by hydrosquall** [reply]
> regular maintainer meeting

I like this idea even if it were not directly tied to the sponsor slots, this seems to be a common practice among other open source projects as a way to get facetime between new/existing contributors without having to schedule dozens of 1:1s.

At minimum we could float the idea in the existing maintainers channel, and see who would be interested to attend.

**Reply by hydrosquall** [reply]
> re-release of a previous version for the first time to address a security issue. Generally want to avoid it in the future.

The LTS (long term support) branch is [here](https://github.com/vega/vega/tree/long-term-support/vega-5) . We could potentially follow the lead of Python/Node of declaring an "end of life" date for this branch after which we'll no longer push security updates here - interested in what others think. 

We did it this time since Vega V6 was the first breaking API change in 5 years with the move to ESM, and some groups were unable to upgrade to the patched version since their application did not support ESM tooling. If the hurdle to upgrading was not build tooling related, the case for doing a backport would have been much weaker (i.e. we are not supporting vega v4, v3 at this point).

**Reply by joelostblom** [reply]
I was reading [the sponsorship page of JSON schema](https://json-schema.org/overview/sponsors) today. I think they clearly convey information and have seemingly reasonable rewards and prices for the different sponsorship tiers. Potentially we could draw some inspiration from how they are doing it.

**Reply by mattijn** [reply]
Good reference @joelostblom! I just notice another thing, at other repos that are registered at numfocus has this type of url for individual donations: [https://numfocus.org/donate-to-gdal](https://numfocus.org/donate-to-gdal), but there is not (yet) such a thing as [https://numfocus.org/donate-to-vega](https://numfocus.org/donate-to-vega)

**Comment by domoritz** [comment]
# Steering Committee Meeting March 3, 2026

## Participants

@jonmmease 
@hydrosquall 
@mattijn 
@kanitw 
@joelostblom 
@domoritz 

## Updates from projects

* Altair: Working through version updates, regression in v6, and a new feature for layering facet charts (for now implemented as a rewrite to facet of layer in Altair but could migrate the logic to Vega-Lite).
* VL-Convert: Big dependency updates 🎉.  Working on V2 with better font support.
* Vega fusion: Contribution from DeepNote for Spark support (getting towards the direction of supporting pushing computation to other databases).
* Vega/Vega-Lite: Vega Utils is now in Typescript. Hardening security and setting up better dev experience. Vega-Lite lasso.
* Prototype: MCP App to have Vega widgets in e.g. Claude. Using Vega Fusion and VL-Convert as backend (to send image to the agent and to bundle an app for the fronted with Vega Embed).
* Vega AI tool with Datadog.

## Bounty

https://github.com/vega/vega/discussions/categories/-experiment-feature-request-fastlane

* How can people sponsor individual features.
* For the sizes (up to 100$) an AI agent could also make the contributions but not at the level of quality that is needed. 
* It's unclear who would pick up issues and who could then review the final pull request to actually finish the work.
* Dom says there is a lot of value in bringing in more contributors. Sometimes that doesn't mean the time invested by core maintainers is less than if they just fixed the big themselves.
* While the Vega project does not directly support the bounty program with financial support, we should promote contributions and highlight people.

## Sponsoring Docs

Draft for the website: https://github.com/vega/vega.github.io/tree/sponsor-page

* Will drop bounty support for now
* Update graduate student to be less specific
* Core-developer summit to conference (link to previous conference)
* Call out to proposals: https://github.com/vega/.github/discussions/25
  * Fund mentoring, tutorials at conferences, etc.
  * Feature development.
* Fund to pay for security fixes, maintenance tasks, etc
* Update to 10k for the partner tier
* remove blog for now
* Partner tier gets mentioned in the release notes for Vega, Vega-Lite, and Altair

Potential candidates we can send this to: Hex, Databricks, Datadog, Elastic.
Next step: PowerBI, Google.

@mattijn @domoritz @jonmmease @kanitw @hydrosquall work on this over the next week to review. 

- [x] Update doc as discussed above (everyone)
- [x] Figure out tiers @mattijn 
- [x] Run the document by NumFocus @domoritz
- [ ] Send it out. 
 
## AI Policy

No explicit policy at the moment but we expect any contributor to be fully responsible for every line of code. For now, we have not seen issues but we may add an explicit policy when there is an issue.
If we can use AI for triaging etc that would be great.

**Comment by domoritz** [comment]
# Steering Committee Meeting May 26, 2026

## Participants

@jheer
@domoritz 
@ChristopherDavisUCI 
@joelostblom
@hydrosquall
@mattijn 

## Updates from projects

6.1.0 release of Altair 🎉! Lots of activity on the Vega/Vega-Lite repos. First bounty received a pull request. We need to consider what to do about agentic PRs. We now have an npm org for Vega. This simplifies adding people to get release permissions. 

## Wins 

* OpenAI is using Vega in its blog: https://openai.com/index/model-disproves-discrete-geometry-conjecture/
* Cameron released an MCP server for Vega MCP Apps: https://github.com/hydrosquall/vega-mcp-server

## Community building (@hydrosquall leading)

* Proposal for per-repo leads (e.g., lead maintainers for core repos)
* Proposal for ADOPTERS.md 
* Proposal for impact / community survey report

### Notes

* Maybe have a flag on each repo whether it's research/beta/production. https://github.com/vega/.github/pull/31/changes/dd0bd89dda3aa2620e6be2e9a6eb1d4c25e3888c..70de2150df4c2d34cde61c6ace679fd764d4ae7a is a step towards that. 
* Having leads and deputies for each project is a great idea
* Support for having alumni/emeritus status
* Could we highlight people in the newsletter? Like "community member of the month".
* Discord was discussed. Needs more community input. 

## Next steps for sponsoring

- [ ] Send out sponsorship call to industry partners.

- Open Source for Science Fund: https://x.com/renphilanthropy/status/2051331642824753336?s=46. Needs a lead to step to to collect evidence and prepare application. 

## Agentic PRs and how to deal with them

Double-edged sword. One one had allows people who are in the community to send PRs for changes. It empowers people and if the PRs are good, it makes sense to adopt the changes. However, AI PRs are often complex, and the author may not fully understand the implementation or whether it could have been simpler. This increases the maintenance burden (which is already challenging). Drive-by PRs (one off PRs) could be a burden when the author does not respond to feedback.

No general policy. Proposed strategy: build of a ladder of engagement. Start with a small reaction (e.g. comment on a pull request asking for motivation), then engage more and more if we see that the author is engaging. 

## TODO

- [ ] @hydrosquall and @domoritz prepare proposal for updating governance docs with Emeritus and Leads. 
- [ ] @hydrosquall will set up proposal for project health documentation.

**Comment by domoritz** [comment]
# Steering Committee Meeting August 25, 2026

## Participants

@domoritz 
@joelostblom 
@hydrosquall 
@kanitw 

## Updates from projects

@domoritz Merged a bunch of PRs in Vega and made a release. Developed a process with git work trees + claude to triage a bunch of PRs and make a release. 

Typescript schema generator is in go. 

Vega-Lite now should run the full tests on forks (previously people had to add a PAT).

Altair had a few fixes and work on automatic releases (with GitHub actions).

## Wins

https://github.com/datamazelab/vl2x as a cool demo.

## Sponsorship

Let's get companies to become partners. 

## Community health

@hydrosquall wrote a proposal for lead maintainers per repo. Lead maintainers can self merge. Core maintainers are trusted and should judge whether they want another set of eyes on API changes. 

@hydrosquall has a few candidates for helping with a newsletter.

We are having great contributions coming in especially from Omni, Databricks, and Microsoft. 

## Security update

In the recent releases, we get AI bots sending safety reports. It takes time to review and often turns out to not be credible security issues. 

## AI contributions

Shut down bug bounty because it led to too many drive by contributions. 

It's been great that community members can contribute to other packages (e.g. Altair contributors to Vega).

Altair has had good experiences.

## VIS mini meetup

Let's do it. Maybe in a break or an evening at the Datadog office?
